File: 문하현/_2_3.java
package week2;


public class _2_3 {
	public static void main(String[] args){
		
		
	}
	
	public String longestCommonPrefix(String[] strs) {
		String answer="";
		if(strs.length == 0) return answer;
		
		answer = strs[0];
		
		for(int i =1;i<strs.length;i++) {
			String cur = strs[i];
			while(cur.indexOf(answer)!=0) {
				answer = answer.substring(0,answer.length()-1);
			}
		}
		
		//indexOf()는 주어진 요소가 배열에 있다면 그 첫번째 인덱스 값을 반환하고 없으면 -1을 반환한다. 
		
        return answer;
    }
}

File: 김현욱/1주차/2_3.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) {
            return "";
        }
        // 첫 번째 단어 길이 만큼 반복한다.
        for (int i = 0; i < strs[0].length(); ++i) {
            // strs에 담긴 배열의 개수만큼 반복한다.
            for (const auto& str : strs) {
                if (i >= str.length() || str[i] != strs[0][i]) {
                    return strs[0].substr(0, i);
                }
            }
        }
        return strs[0];
    }
};